catalog/core/catalog-core-camelcomponent/src/main/java/ddf/camel/component/catalog/content/ContentProducerDataAccessObject.java

```diff
@@ -213,4 +222,21 @@ public void processHeaders(Map<String, Object> headers, StorageRequest storageRe
                     .put(Constants.STORE_REFERENCE_KEY, ingestedFile.getAbsolutePath());
         }
     }
+
+    private void waitForAvailableSource(CatalogFramework catalogFramework)
+            throws SourceUnavailableException {
+        try {
+            WaitCondition.expect("Source should be available")
+                    .checkEvery(3, TimeUnit.SECONDS)
+                    .within(3, TimeUnit.MINUTES)
+                    .until(() -> catalogFramework.getSourceInfo(new SourceInfoRequestLocal(false))
+                            .getSourceInfo()
+                            .stream()
+                            .findFirst()
+                            .get()
```

Comment:
I could be wrong, but I think `get()` will throw an `NoSuchElementException` if `getSourceInfo()` returns an empty set. If that is the case, you can use `ifPresent()` instead of `get()`